ethstorage/es-node
# fix Issue74: use NAT ips as fallback ips if AdvertiseIP no be set

related issue: https://github.com/ethstorage/es-node/issues/74

If the users do not assign their public IP address in the command line flag and start es-node, it will not provide p2p sync service to other peers because the generated ENR doesn't contain the IP address. 
As the node's host enables NAT service, we can use the host's addresses in the ENR if the AdvertiseIP is not set. 

How to verify:
Run the node without AdvertiseIP, and view the ENR at https://enr-viewer.com/. and see the IP in tcp4Multiaddr.
